driver-core/src/main/com/mongodb/client/model/Filters.java

```diff
@@ -422,6 +422,94 @@ public static Bson text(final String search, final String language) {
     }
 
     /**
+     * Creates a filter that matches all documents matching the given the search term with the given case sensitive flag.
+     *
+     * @param search        the search term
+     * @param caseSensitive the case sensitive search flag, for use with version 3 of the text index.
+     * @return the filter
+     * @mongodb.driver.manual reference/operator/query/text $text
+     * @since 3.2
+     * @mongodb.server.release 3.2
+     */
+    public static Bson text(final String search, final boolean caseSensitive) {
```

Comment:
Given all these new overloads, and the possibility of yet more in the future, I'd prefer that we introduce a TextSearchOptions class.